nemo/core/connectors/save_restore_connector.py

```diff
@@ -146,20 +146,67 @@ def restore_from(
                 if app_state.model_parallel_size is not None and app_state.model_parallel_size > 1:
                     model_weights = self._inject_model_parallel_rank_for_ckpt(tmpdir, self.model_weights_ckpt)
                 state_dict = self._load_state_dict_from_disk(model_weights, map_location=map_location)
-
-                if conf.get('megatron_amp_O2', False):
-                    new_state_dict = {}
-                    for key in state_dict.keys():
-                        new_key = key.replace('model.', 'model.module.', 1)
-                        new_state_dict[new_key] = state_dict[key]
-                    state_dict = new_state_dict
-                instance.load_state_dict(state_dict, strict=strict)
-
-                logging.info(f'Model {instance.__class__.__name__} was successfully restored from {restore_path}.')
-                instance._set_model_restore_state(is_being_restored=False)
             finally:
                 os.chdir(cwd)
 
+        return (conf, instance, state_dict)
+
+    def modify_state_dict(self, conf, state_dict):
```

Comment:
Why is megatron related things in the core implementation of SaveRestoreConnector? Doesnt NLPModel use its own SaveRestoreConnector implementation ?

Reply:
The megatron Bert-related implementation is present in NLP model's saverestore connector, this is megatron GPT which is already present in the core(approved and developed in previous releases/changes).
Yes the correct thing is to move it over to NLPmodel, since I wasn't sure of why it has been approved in the first place I didn't move this to NLPmode;

Reply:
Yes we need to move this out of core, but we can do it for 1.9 I think

Reply:
Sure that's fine